### PR TITLE
upgrading Jetty and XMLUnit as needed to support upgrading the REST A…

### PR DESCRIPTION
**Pull request by vickdw**
…PI to Spring MVC 4

### CODE REVIEW

File: geowebcache/rest/pom.xml [hunk 1]
```diff
@@ -102,8 +102,9 @@
       <classifier>tests</classifier>
     </dependency>
     <dependency>
-      <groupId>xmlunit</groupId>
-      <artifactId>xmlunit</artifactId>
+      <groupId>org.xmlunit</groupId>
+      <artifactId>xmlunit-legacy</artifactId>
+      <version>2.3.0</version>
```
Comment (smithkm) [inline geowebcache/rest/pom.xml]: The version should really be up in the root pom.

File: geowebcache/web/src/test/java/org/geowebcache/jetty/JettyRule.java [hunk 2]
```diff
@@ -1,31 +1,31 @@
...
-import org.junit.rules.TemporaryFolder;
-import org.junit.runner.Description;
-import org.junit.runners.model.Statement;
-import org.mortbay.jetty.Connector;
-import org.mortbay.jetty.Server;
-import org.mortbay.jetty.bio.SocketConnector;
-import org.mortbay.jetty.webapp.WebAppContext;
-import org.mortbay.thread.QueuedThreadPool;
+//import org.eclipse.jetty.bio.SocketConnector;
+//import org.eclipse.thread.QueuedThreadPool;
```
Comment (smithkm) [inline geowebcache/web/src/test/java/org/geowebcache/jetty/JettyRule.java]: Commented out code shouldn't be left in without a good reason.

File: geowebcache/web/pom.xml [hunk 3]
```diff
@@ -98,7 +98,7 @@
       <groupId>org.springframework.security</groupId>
       <artifactId>spring-security-config</artifactId>
     </dependency>
-    
+
```
Comment (smithkm) [inline geowebcache/web/pom.xml]: Non-semantic whitespace changes

File: geowebcache/web/src/test/java/org/geowebcache/jetty/JettyRule.java [hunk 4]
```diff
@@ -1,31 +1,31 @@
...
+ *
  * @author Kevin Smith, Boundless
  *
  */
 public class JettyRule extends org.junit.rules.ExternalResource {
-    
+
     Integer port = null;
-    
+
```
Comment (smithkm) [inline geowebcache/web/src/test/java/org/geowebcache/jetty/JettyRule.java]: Non-semantic whitespace changes.  Lots of them in this file.

File: geowebcache/web/src/test/java/org/geowebcache/jetty/JettyRule.java [hunk 5]
```diff
@@ -74,54 +74,61 @@ public JettyRule(Initializer<File> confInit, Initializer<File> cacheInit) {
...
+//                conn.setPort(port);
+//                try{
+//                    conn.open();
+//                } catch (BindException e) {
+//                    continue;
+//                }
+//            }
+//            conn.setPort(port);
+//            conn.setAcceptQueueSize(100);
+//            jettyServer.setConnectors(new Connector[] { conn });
```
Comment (smithkm) [inline geowebcache/web/src/test/java/org/geowebcache/jetty/JettyRule.java]: Commented out code.

File: geowebcache/web/src/test/java/org/geowebcache/jetty/JettyRule.java [hunk 6]
```diff
@@ -131,13 +138,13 @@ protected void before() throws Exception {
...
             wah.setTempDirectory(workDir);
-            
+
             // Use this to set a limit on the number of threads used to respond requests
             QueuedThreadPool tp = new QueuedThreadPool();
             tp.setMinThreads(50);
             tp.setMaxThreads(50);
-            conn.setThreadPool(tp);
-            
+//            conn.setThreadPool(tp);
```
Comment (smithkm) [inline geowebcache/web/src/test/java/org/geowebcache/jetty/JettyRule.java]: Commented out code

File: geowebcache/web/src/test/java/org/geowebcache/jetty/Start.java [hunk 7]
```diff
@@ -1,24 +1,25 @@
...
-import org.mortbay.jetty.bio.SocketConnector;
-import org.mortbay.jetty.webapp.WebAppContext;
-import org.mortbay.thread.QueuedThreadPool;
-
 /**
  * Jetty starter, will run GeoWebCache inside the Jetty web container.<br>
  * Useful for debugging, especially in IDE were you have direct dependencies between the sources of
  * the various modules (such as Eclipse).
- * 
+ *
```
Comment (smithkm) [inline geowebcache/web/src/test/java/org/geowebcache/jetty/Start.java]: Whitespace

File: geowebcache/web/src/test/java/org/geowebcache/jetty/Start.java [hunk 8]
```diff
@@ -27,28 +28,38 @@ public static void main(String[] args) {
...
+            http.setSoLingerTime(-1);
+
+//            SocketConnector conn = new SocketConnector();
+//            String portVariable = System.getProperty("jetty.port");
+//            int port = parsePort(portVariable);
+//            if (port <= 0)
+//                port = 8080;
+//            conn.setPort(port);
+//            conn.setAcceptQueueSize(100);
+//            jettyServer.setConnectors(new Connector[] { conn });
```
Comment (smithkm) [inline geowebcache/web/src/test/java/org/geowebcache/jetty/Start.java]: Commented code

File: geowebcache/web/src/test/java/org/geowebcache/jetty/JettyRule.java (hunk 4 above)
Comment (vickdw) [inline geowebcache/web/src/test/java/org/geowebcache/jetty/JettyRule.java]: Yes, I'm not sure how they got in there.  Might be an artifact of copy/paste between applications.

**Pull request merged**